test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetPackCommandTest.cs

```diff
@@ -2164,6 +2164,191 @@ public void PackCommand_ReferencedProjectWithCustomTokensInNuspec()
+        {
+            // Arrange
+            const string version = "1.2.3.4";
+            const string informationalVersion = "3.4.5.6 invalid";
+
+            var nugetexe = Util.GetNuGetExePath();
+            using (var workingDirectory = TestDirectory.Create())
+            {
+                CreateTestProjectWithAssemblyInfoAndNuspec(workingDirectory, "Foo", version, "2.3.4.5", informationalVersion);
+                var projectDirectory = Path.Combine(workingDirectory, "Foo");
+
+                // Act
+                var r = CommandRunner.Run(nugetexe, projectDirectory, "pack Foo.csproj -build", waitForExit: true);
+
+                // The assembly version was used, not the informational version
+                var outputPackageFileName = Path.Combine(projectDirectory, $"Foo.{version}.nupkg"); 
+                var outputPackage = new OptimizedZipPackage(outputPackageFileName);
+
+                // Assert
+                Assert.True(0 == r.Item1, r.Item2 + " " + r.Item3); // Exited successfully
+                Assert.Equal(new SemanticVersion(version), outputPackage.Version);
+            }
+        }
+
+        [Fact]
+        public void PackCommand_NuspecFileWithTokens_uses_informational_version() // @odalet, https://github.com/NuGet/Home/issues/5548
+        {
+            // Arrange
+            const string version = "1.2.3.4";
+            const string fileVersion = "2.3.4.5";
+            const string semverVersion = "3.4.5.6-beta.1";
+            const string informationalVersion = semverVersion + "+additional-info";
+
+            var nugetexe = Util.GetNuGetExePath();
+            using (var workingDirectory = TestDirectory.Create())
+            {
+                CreateTestProjectWithAssemblyInfoAndNuspec(workingDirectory, "Foo", version, fileVersion, informationalVersion);
+                var projectDirectory = Path.Combine(workingDirectory, "Foo");
+
+                // Act
+                var r = CommandRunner.Run(nugetexe, projectDirectory, "pack Foo.csproj -build", waitForExit: true);
+
+                // The informational version without the build metadata part was used
+                var outputPackageFileName = Path.Combine(projectDirectory, $"Foo.{semverVersion}.nupkg"); 
+                var outputPackage = new OptimizedZipPackage(outputPackageFileName);
+
+                // Assert
+                Assert.True(0 == r.Item1, r.Item2 + " " + r.Item3); // Exited successfully
+                Assert.Equal(new SemanticVersion(semverVersion), outputPackage.Version);
+            }
+        }
+
+        // This test is a bit redundant with the ones before, but still useful for debugging
+        [Fact]
+        public void PackCommandRunner_does_not_throw_when_informational_version_is_invalid() // @odalet, https://github.com/NuGet/Home/issues/5548
```

Comment:
The naming should be PascalCase + snake case in the form of "What_UnderWhatConditions_Expectation"

So `PackCommandRunner_WhenInformationalVersionIsInvalid_DoesNotThrow`